ctags/main/ctags-api.c

```diff
@@ -0,0 +1,141 @@
+/*
+*   Copyright (c) 2016, Jiri Techet
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your option) any later version.
+*
+*   Defines ctags API when compiled as a library.
+*/
+
+#include "general.h"  /* must always come first */
+
+#ifdef CTAGS_LIB
+
+#include "ctags-api.h"
+#include "types.h"
+#include "routines.h"
+#include "error.h"
+#include "output.h"
+#include "parse.h"
+#include "options.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+static bool nofatalErrorPrinter (const errorSelection selection,
+					  const char *const format,
+					  va_list ap, void *data CTAGS_ATTR_UNUSED)
+{
+	fprintf (stderr, "%s: ", (selection & WARNING) ? "Warning: " : "Error");
+	vfprintf (stderr, format, ap);
+	if (selection & PERROR)
+#ifdef HAVE_STRERROR
+	fprintf (stderr, " : %s", strerror (errno));
+#else
+	perror (" ");
```

Comment:
Yeah, we'd need some proper API for that which we don't have. The problem I was running into was that in tag definitions of individual parsers some tags were disabled by default and "make check" failed because of that. So it was either disabling the enabled check here or changing all the affected parsers tag definitions and this was just easier.